### ISSUE TITLE
plot_cnv quantile.default missing values and NaN's not allowed if 'na.rm' is FALSE

### ISSUE THREAD

**Issue by GHAStVHenry** [broadinstitute/infercnv#327]
running `infercnv::run` with `plot_steps=TRUE` leads to:
```
Error in quantile.default(plot_data[plot_data != x.center], c(0.01, 0.99)) :
  missing values and NaN's not allowed if 'na.rm' is FALSE
```

Full output:
```
INFO [2021-05-17 16:30:54] ::order_reduce:Start.
INFO [2021-05-17 16:31:00] .order_reduce(): expr and order match.
INFO [2021-05-17 16:31:05] ::process_data:order_reduce:Reduction from positional data, new dimensions (r,c) = 36601,199350 Total=780116943 Min=0 Max=29791.
INFO [2021-05-17 16:31:06] num genes removed taking into account provided gene ordering list: 1280 = 3.49717220840961% removed.
INFO [2021-05-17 16:31:08] -filtering out cells < 100 or > Inf, removing 1.81741 % of cells
INFO [2021-05-17 16:31:31] validating infercnv_obj
             used   (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells    6627052  354.0   11865068   633.7    8664836   462.8
Vcells 1085886163 8284.7 3144512212 23990.8 3860337236 29452.1
INFO [2021-05-17 16:31:32] ::process_data:Start
INFO [2021-05-17 16:31:32] Checking for saved results.
INFO [2021-05-17 16:31:32]

        STEP 1: incoming data

INFO [2021-05-17 16:36:46]

        STEP 02: Removing lowly expressed genes

INFO [2021-05-17 16:36:46] ::above_min_mean_expr_cutoff:Start
INFO [2021-05-17 16:37:03] Removing 31060 genes from matrix as below mean expr threshold: 0.1
INFO [2021-05-17 16:37:14] validating infercnv_obj
INFO [2021-05-17 16:37:14] There are 4261 genes and 195727 cells remaining in the expr matrix.
INFO [2021-05-17 16:38:08] no genes removed due to min cells/gene filter
INFO [2021-05-17 16:42:35]

        STEP 03: normalization by sequencing depth

INFO [2021-05-17 16:42:35] normalizing counts matrix by depth
INFO [2021-05-17 16:43:54] Computed total sum normalization factor as median libsize: 1921.000000
INFO [2021-05-17 16:43:55] Adding h-spike
INFO [2021-05-17 16:43:55] -hspike modeling of Normal_D17PrPzF_Via
INFO [2021-05-17 16:48:49] -hspike modeling of Normal_D17PrTzF_Via
INFO [2021-05-17 16:53:35] -hspike modeling of Normal_D27PrPzF_Via
INFO [2021-05-17 16:58:13] -hspike modeling of Normal_D27PrTzF_Via
INFO [2021-05-17 17:02:48] -hspike modeling of Normal_D35PrPzF_Via
INFO [2021-05-17 17:07:22] -hspike modeling of Normal_D35PrTzF_Via
INFO [2021-05-17 17:11:57] validating infercnv_obj
INFO [2021-05-17 17:11:57] normalizing counts matrix by depth
INFO [2021-05-17 17:11:57] Using specified normalization factor: 1921.000000
INFO [2021-05-17 17:16:10]

        STEP 04: log transformation of data

INFO [2021-05-17 17:16:10] transforming log2xplus1()
INFO [2021-05-17 17:16:50] -mirroring for hspike
INFO [2021-05-17 17:16:50] transforming log2xplus1()
INFO [2021-05-17 17:21:16] ::plot_cnv:Start
INFO [2021-05-17 17:21:16] ::plot_cnv:Current data dimensions (r,c)=4261,195727 Total=201572061.704079 Min=0 Max=10.857345485359.
INFO [2021-05-17 17:21:20] ::plot_cnv:Depending on the size of the matrix this may take a moment.
Error in quantile.default(plot_data[plot_data != x.center], c(0.01, 0.99)) :
  missing values and NaN's not allowed if 'na.rm' is FALSE
Calls: <Anonymous> -> plot_cnv -> quantile -> quantile.default
In addition: Warning message:
In mean.default(infercnv_obj@expr.data) :
  argument is not numeric or logical: returning NA
Execution halted
```

**Comment by GeorgescuC** [comment]
Hi @GHAStVHenry ,

What version of infercnv are you using? It seems you have a large number of cells, so if you plan on using the HMM/subclustering, I would very highly recommend using the latest version (1.8.0) if you are not already as there is a new method for subclustering that is much faster.

It looks like there is an issue some values in the matrix. Can you double check that the input you provide has no NA/NaN values? If it does not, can you also load the backup objects from steps 3 and 4 (with `readRDS()`) and see if any NA/NaN appeared in the `infercnv_obj@expr.data` field in either?

Regards,
Christophe.

**Comment by GHAStVHenry** [comment]
just checked, I'm using 1.6.0... I didn't even realize it's outdated... I'll update and retry...

**Comment by GHAStVHenry** [comment]
👎 docker container does not contain 1.8.0 and I don't have R 4.1.0 on my HPC (and of course I don't have sudo to install it)

**Comment by GHAStVHenry** [comment]
I just double checked and it doesn't appear that infercnv 1.8.0 actually required R 4.1.0... even though bioconductor is requiring it for the install... I guess I can install from github... will try that

**Comment by GHAStVHenry** [comment]
Confirmed... installed from source... will test

**Comment by sunshine1126** [comment]
@GeorgescuC Hello, I got the same problem when I set the parameter `plot_steps=TRUE` in the version infercnv 1.8.0. Also, I check if any NA/NaN appeared in the infercnv_obj@expr.data the backup objects from steps 3 and 4, but it showed no NA/NaN values. My command is as follows.

> infercnv_obj = infercnv::run(infercnv_obj,
>                              cutoff=0.1,
>                              out_dir="infercnv_dir_noise_filtered",
>                              cluster_by_groups=FALSE,
>                              plot_steps=T,
>                              scale_data=T,
>                              denoise=T,
>                              noise_filter=0.12,
>                              analysis_mode='subclusters',
>                              tumor_subcluster_partition_method = "random_trees",
>                              HMM=TRUE,
>                              HMM_type="i6",
>                              num_threads=10)
> infercnv_obj_1 <- readRDS("~/infercnv_dir_noise_filtered/04_logtransformedHMMi6.infercnv_obj")
> sum(is.na(infercnv_obj_1@expr.data)) ## 0
>

**Comment by GeorgescuC** [comment]
Hi @sunshine1126 ,

The method `is.na()` only checks for NA, but there can be other problematic types of values. Could you try the following `all(is.finite(infercnv_obj_2@expr.data))` and make sure it returns TRUE?
If it does, would you be able to share the data privately so I can look into the issue?

Regards,
Christophe.

**Comment by sunshine1126** [comment]
@GeorgescuC Thanks for your reply. After running the command `all(is.finite(infercnv_obj_2@expr.data))`, it indeed returned TRUE.
![image](https://user-images.githubusercontent.com/32659317/133362161-6f35da2a-10aa-4ec2-b1d1-2e4f30f2965e.png)
Can you give me your email that I sent the data to you?

**Comment by GeorgescuC** [comment]
Hi @sunshine1126 ,

You can find my email in the authors list of [the description file](https://github.com/broadinstitute/infercnv/blob/master/DESCRIPTION) .

Regards,
Christophe.

**Comment by sunshine1126** [comment]
@GeorgescuC Thanks for your reply and I will send it to you.

**Comment by adhisadi** [comment]
I am getting the same problem with version 1.10.0 
Also all(is.finite(infercnv_obj@expr.data)) returns
[1] TRUE

How was this issue solved?

**Comment by GeorgescuC** [comment]
Hi @adhisadi ,

For this case, the issue was due to the data being in sparse matrix format and the plot_cnv method not supporting it. This only happened if using "plot_steps=T" as the matrix is not sparse anymore after the smoothing step.
This issue is fixed in version 1.11.1 of infercnv (available in BioConductor Devel).

Regards,
Christophe.

**Comment by adhisadi** [comment]
Thank you for your response, I might as well try it with version 1.11.1.
